lib/razor/config.rb

```diff
@@ -54,19 +54,23 @@ def initialize(env, fname = nil, defaults_file = nil)
               _("The configuration defaults file %{filename} is not readable") % {filename: defaults_file}
       end
 
-      begin
-        yaml = File::open(fname, "r") { |fp| YAML::load(fp) } || {}
-
-        @values.merge!(yaml["all"] || {})
-        @values.merge!(yaml[Razor.env] || {})
-      rescue Errno::ENOENT
-        # This is only an error if we don't have any config values yet.
-        raise InvalidConfigurationError,
-          _("The configuration file %{filename} does not exist") % {filename: fname} if @values.empty?
-      rescue Errno::EACCES
-        raise InvalidConfigurationError,
-          _("The configuration file %{filename} is not readable") % {filename: fname}
+      if File.readable?(fname)
+        begin
+          yaml = File::open(fname, "r") { |fp| YAML::load(fp) } || {}
+
+          @values.merge!(yaml["all"] || {})
+          @values.merge!(yaml[Razor.env] || {})
+        rescue Errno::ENOENT
+          # This is only an error if we don't have any config values yet.
+          raise InvalidConfigurationError,
+            _("The configuration file %{filename} does not exist") % {filename: fname} if @values.empty?
```

Comment:
I wonder if this is something we want to raise outside of the if? As in, do we need to have a configuration before returning from the initialize method in the config.rb class or only when we know that fname exits?

Reply:
Let's just nix the `readable?` if block here. The `if @values.empty?` bit catches the main concern here.